firebase-crashlytics/src/main/java/com/google/firebase/crashlytics/internal/DevelopmentPlatformProvider.java

```diff
@@ -59,6 +53,65 @@ public String getDevelopmentPlatform() {
    */
   @Nullable
   public String getDevelopmentPlatformVersion() {
-    return developmentPlatformVersion;
+    return initDevelopmentPlatform().developmentPlatformVersion;
+  }
+
+  /**
+   * Returns if the development platform is Unity, without initializing the rest of the
+   * DevelopmentPlatform object.
+   *
+   * <p>This is useful for the NDK to avoid an expensive file operation during start up.
+   */
+  public static boolean isUnity(Context context) {
+    return getResourcesIdentifier(context, UNITY_VERSION_FIELD, "string") != 0;
+  }
+
+  /** Quickly and safely check if the given asset path exists. */
+  private boolean assetPathExists(String path) {
+    try {
+      if (context.getAssets() == null) {
+        return false;
+      }
+      String[] list = context.getAssets().list(path);
+      return list != null && list.length > 0;
+    } catch (IOException ex) {
+      return false;
+    }
+  }
+
+  private DevelopmentPlatform initDevelopmentPlatform() {
```

Comment:
Does this method need to be synchronized?

Reply:
I think it's ok to not synchronize it. Because even if we have two instances, it's read-only so nothing bad will happen.